tests/main/hidden-snap-dir/task.yaml

```diff
@@ -184,3 +184,22 @@ execute: |
     check_env "--with-exposed-home" "x3"
     check_dirs "--with-exposed-home" "x3"
     check_data "--with-exposed-home" "x3" "$data"
+
+    echo "Check migration after remove works"
+    # ensure dirs under ~/.snap/data are created
+    "$NAME".cmd 'true'
+
+    echo "Remove snap and check dir was cleaned up"
+    snap remove --purge "$NAME"
+    not test -e "$HOME/.snap/data/$NAME"
+
+    echo "Install snap under ~/snap"
+    snap pack "$TESTSLIB/snaps/$NAME"
+    snap unset system experimental.hidden-snap-folder
+    "$TESTSTOOLS"/snaps-state install-local "$NAME"
+    # create dirs under ~/snap to be migrated
+    "$NAME".cmd 'true'
+
+    echo "Refresh w/ flag (migrate to ~/.snap/data)"
+    snap set system experimental.hidden-snap-folder=true
+    "$TESTSTOOLS"/snaps-state install-local "$NAME"
```

Comment:
It would be nice to check that `"$HOME/snap/$NAME"` does not exist and that `"$HOME/.snap/data/$NAME"` exists.

Reply:
+1